widget/IntersectionBetweenModelErrors.tsx

```diff
@@ -89,7 +90,10 @@ function calculateCircleRadiiAndDistance(a, b, ab, datasetSize) {
 
 type IntersectionBetweenModelErrorsState = {
   selectedDataPoint: any,
-  regionSelected: any
+  regionSelected: any,
+  progressFill: string,
+  regressFill: string,
+  intersectionFill: string
```

Comment:
Do we really need these values of the fills in here? Can they not be derived from the `regionSelected` and perhaps other state information?

I'd suggest not directly putting style information within the state. Instead state should conceptually be about the logical state of the component ... i.e. what region is selected or what data is being shown etc. There should then be functions that we can pass that info into which will then derive style properties like fill color etc.